include/eblob/blob.h

```diff
@@ -690,6 +690,14 @@ int eblob_start_defrag(struct eblob_backend *b);
  */
 int eblob_start_defrag_level(struct eblob_backend *b, enum eblob_defrag_state level);
 
+/*!
+ * eblob_start_defrag_in_dir() - overload version of eblob_start_defrag with defrag-level and chunks_dir params.
+ * @b - eblob backend instance
+ * @level - level of defragmentation
+ * @chunks_dir - directory for temporary chunks (overrides value specified in config)
+ */
+int eblob_start_defrag_in_dir(struct eblob_backend *b, enum eblob_defrag_state level, const char *chunks_dir);
```

Comment:
Just an idea. We could develop unify flexible interface, like in `dnet_access_context` in C, to specify any params you want to handler. So we won't have to create a lot of alternate signatures.
```
params = create_params();
params_add_int(params, "level", 1);
params_add_int(params, "chunks_dir", "/tmp/chunks");
eblob_start_defrag_new(b, params);
```

Reply:
`dnet_access_context` is a dynamic key-value map that is printed into log as is.

Using key-value dynamic map for passing arguments into a function turns off compile time type checking and may hide typos in key. I would prefer using `defrag_config` structure with reserved fields.

Reply:
Yes, your variant is more reliable (although slightly less flexible). In WinAPI, I remember, all structs have field `.cbSize` instead of using reserved bytes. I see, reserved is more traditional in the project. Ok.